### ISSUE TITLE
Uneekor Putting Null  Fields Give Errors in Connector

### ISSUE THREAD

**Issue by kheninge** [springbok/MLM2PRO-GSPro-Connector#152]
When  you putt with the Uneekor View system in putting mode it only reports Ball Speed, Side Angle and Spin Axis. All of the other values get "--". The connector seems to either miss the putt altogether or it interprets the Spin Rate -- as 9999999. Also the "--" for Back Spin and Launch Angle get reported as errors.

When I have tried putting with Uneekor in normal mode either the Uneekor system itself misses the putt or it often reports one of the fields above as "--" which leads to an error. About 30% of the time the putt will get picked up and reported by the connector.

I'm attaching a couple of photos to show the relevant OCR screen and the error reported. BTW I have tried this with both View and Multi View and it both have the same issue.
![PXL_20241206_042645857](https://github.com/user-attachments/assets/a9c2ab94-20c9-490b-8188-3d3dbcbe0615)
![PXL_20241206_042547164 MP](https://github.com/user-attachments/assets/42269a05-40f8-4bb0-a799-b0083c40ae67)

**Comment by kheninge** [comment]
Ok, doing some more testing. It looks like I was getting errors only because the selected club in GSPro was SW. When putter is selected then the Uneekor connector is Paused so the Uneekor putting inputs are ignored. So does that mean that putting in Uneekor is not currently supported? Or am I missing an options somewhere?

**Comment by wonder99** [comment]
I'll investigate. I confess, I added the uneekor support but didn't test uneekor putting because I use exputt, so that and webcam putting should work fine.

**Comment by wonder99** [comment]
Please download this and let me know how it goes. For Uneekor and with Putting System set to None, you should be able to putt with Uneekor.
https://www.dropbox.com/scl/fi/wsrqb02vg6obj3ca7h44x/MLM2Pro-GSPro-Connector.exe?rlkey=9mum04prbzi2dlrbhnlkg927y&st=fh1yjbwr&dl=1

After you change your putting system, it's easiest to restart the connector, though depending on the state of GSPRO, there are ways to keep the connector running.

**Comment by kheninge** [comment]
Thanks for the quick turnaround this. Your fix solves the problem with pausing during putting. Now the Uneekor View OCR runs and grabs numbers for those putts. The problem now is that a VLA of 0.0 or "--" will cause the connector to error (see previous attached picture). 

I think that when putter is selected, Putting System is set to None and Uneekor is the system the connector should ignore all fields other than HLA (i.e. side angle) and ball speed. Those are the two fields along with Spin Axis are the only fields Uneekor provides, but Spin Axis is always set to 0.0.   I'm not sure what if anything GSPro does with any of the fields other than those 2, but maybe you have some insight from the ExPutt mode?

**Comment by wonder99** [comment]
Well, my plan was to not have the user select Putting mode in View.  Just leave it in full shot mode.  Does that work for you?

**Comment by kheninge** [comment]
I tried it in both modes. It fails 100% in putting mode, and about 50% of the time in full shot mode because the VLA gets reported as 0.0.  From my brief testing it seems like the putting mode may be necessary for shorter putts, but I will experiment further and report back. 

My preference would be to stay in full shot mode as well since I don't have a good solution for changing modes easily. If it turns out that putting mode is needed I am going to experiment with an autohotkey script to change modes programatically.

**Comment by wonder99** [comment]
Sounds good. Stay tuned

**Comment by wonder99** [comment]
Ok I have something working, except I don't see how to switch to putting
mode in View--If I switch to a session that was configured for putting, it
auto-replays the last shot from that session.  I haven't found any other
way to tell View to expect a putt, other than switching sessions.  Is there
a way?

**Comment by kheninge** [comment]
Switching to another session is the only way I have been able to get it into the putting mode. And I didn't think about the problem you are seeing, but that makes sense. I guess the only way to do it would be to create a new session and then delete it afterward or delete the shots after every putt. Either way it would be painful.

The good news is that from my testing the normal mode seems to pick up all of the putts no matter what speed and the speed seems just as accurate as putting mode. I still have some questions about HLA, but quite frankly the HLA seems pretty bad in either mode. (I'm kind of disappointed since putting is one of the things I was hoping would be better than my SkyTrak. But it seems about the same)

**Comment by wonder99** [comment]
Ya, I haven't been very happy with putting accuracy with EML, both
regarding HLA and even speed.
I prefer to stick with my exputt, which is perfect for me.

Also, I found some putts would be ignored if EML was not in putting mode.

I wonder if the 3rd party connector switches modes automatically based on
GSpro. I assume it does, and that's not something we can replicate,
unfortunately.

**Comment by kheninge** [comment]
Yes, I'm pretty sure that Uneekor connector puts it into putting mode and maybe even modes for chipping and driving. But I don't see any way to do that other then messing with sessions and I don't think View even provides any hotkeys, so it would all have to be mouse clicks which is going to be pretty flakey.

**Comment by wonder99** [comment]
Based on that, do you think it's worth trying to support Uneekor putting with this connector?  I'm not so sure.

**Comment by kheninge** [comment]
I think it would only make sense if we could find a way to put the unit into putting mode programmatically. My first attempt failed. When I have some time I may take another shot at it and if I can get it working I'll reach out.

**Comment by khthe8th** [comment]
Just ran into this trying to switch to this connector for my EML. Is there no way to get putting to work with this connector and an EML? Everything else seems to work great, but can't get a single putt to read. I tried to create a new "Putter" session in VIEW but that didn't seem to help either
